docs/input/docs/upgrading/1.x-to-2.x.md

```diff
@@ -0,0 +1,211 @@
+
+### Ensure AppVeyor build image is using latest
+
+In order to execute correctly on AppVeyor, you will need to use the Visual Studio 2019 build image.  Change the image definition from:
+
+```yml
+image: Visual Studio 2017
+```
+
+to:
+
+```yml
+image: Visual Studio 2019
+```
+
+### Ensure that Environment.SetVariableNames() is called
+
+Previously, the call to [Environment.SetVariableNames](../fundamentals/set-variable-names) was not mandatory, and could be omitted from the recipe.cake file.  This is no longer the case, and it must be included.
+
+## Suggested Changes
+
+### Use embedded icon for NuGet package
+
+This guidance comes from the NuGet team directly.  Going forward, you should look to include an icon within the nupkg file.
+
+### Add CakeContrib.Guidelines Nuget Package
+
+To help with the above suggestion, and likely others given time, we are recommending that you add the [CakeContrib.Guidelines](https://www.nuget.org/packages/CakeContrib.Guidelines/) NuGet package to your project.  This will inspect your project, and make sure that you are following the guidelines from the Cake-Contrib Organisation for your addin.
+
+### Remove usage of nuspec file(s) for NuGet packages
+
+Due to the fact that Cake.Recipe has now enabled fully deterministic builds, as well as turning on of SourceLink and embedding pdb files into the generated nupkg file, it is simply easier to run `dotnet pack` rather than attempting to do this manually.  To faciliate this, you will need to add the following parameter:
+
+```csharp
+shouldRunDotNetCorePack: true
+```
+
+### Add Cake.Addin.Analyzer package
+
+Similar to the CakeContrib.Guidelines NuGet Package, we are also recommending that you add the [Cake.Addin.Analyzer](https://www.nuget.org/packages/Cake.Addin.Analyzer/) package to your project.  This will check through your projects and ensure that you are following best practices from a Cake perspective.
+
+### Add settings.json file for VSCode
+
+To help with ensuring consistency of an addins codebase, the following settings.json file is recommended:
+
+```json
+{
+  "omnisharp.enableEditorConfigSupport": true,
+  "omnisharp.enableRoslynAnalyzers": true,
+  "[powershell]": {
+    "files.encoding": "utf8bom"
+  },
+  "powershell.codeFormatting.addWhitespaceAroundPipe": true
+}
+```
+
+### Add dependabot config file
+
+In order to keep your codebase up to date, we recommend that you enable dependabot to help with keeping project dependencies up to date.  This can be done by creating a `.github/dependabot.yml` file in your project with the following contents.
```

Comment:
```suggestion
To keep your dependencies up to date, we recommend that you enable dependabot to help with that task.  This can be done by creating a `.github/dependabot.yml` file in your project with the following contents.
```

Reply:
:+1: